str/runners/str_iterative_hipstr_runner_sharded.py

```diff
@@ -0,0 +1,172 @@
+)
+@click.option('--job-memory', help='Memory of the Hail batch job', default='highmem')
+@click.option(
+    '--variant-catalog',
+    help='Full path to HipSTR Variants sharded catalog directory, and file prefix',
+)
+@click.option('--dataset', help='dataset eg tob-wgs')
+@click.option('--num-shards', help='Number of catalog shards', default=1)
+@click.argument('internal-cpg-ids', nargs=-1)
+@click.option('--output-file-name', help='Output file name without file extension')
+@click.command()
+def main(
+    job_storage,
+    job_memory,
+    num_shards,
+    variant_catalog,
+    dataset,
+    internal_cpg_ids,
+    output_file_name,
+):  # pylint: disable=missing-function-docstring
+    b = get_batch()
+    ref_fasta = 'gs://cpg-common-main/references/hg38/v0/Homo_sapiens_assembly38.fasta'
+    # Read in reference
+    ref = b.read_input_group(
+        **dict(
+            base=ref_fasta,
+            fai=ref_fasta + '.fai',
+            dict=ref_fasta.replace('.fasta', '').replace('.fna', '').replace('.fa', '')
+            + '.dict',
+        )
+    )
+
+    cramfuse_path = get_cloudfuse_paths(dataset, internal_cpg_ids)
+
+    for i in range(int(num_shards)):
+        # Create HipSTR job
+        hipstr_job = b.new_job(name=f'HipSTR shard {i+1} running')
+        hipstr_job.image(HIPSTR_IMAGE)
+        hipstr_job.storage(job_storage)
+        hipstr_job.cpu(4)
+        hipstr_job.memory(job_memory)
+
+        hipstr_job.declare_resource_group(
+            hipstr_output={
+                'vcf.gz': '{root}.vcf.gz',
+                'viz.gz': '{root}.viz.gz',
+                'log.txt': '{root}.log.txt',
+            }
+        )
+        hipstr_job.cloudfuse(f'cpg-{dataset}-main', '/cramfuse')
+
+        # Read in HipSTR variant catalog
+        variant_catalog_input = f'{variant_catalog}_shard_{i+1}.bed'
+        hipstr_regions = b.read_input(variant_catalog_input)
+
+        hipstr_job.command(
+            f"""
+        HipSTR --bams {cramfuse_path} \\
+            --fasta {ref.base} \\
+            --regions {hipstr_regions} \\
+            --str-vcf {hipstr_job.hipstr_output['vcf.gz']} \\
+            --viz-out {hipstr_job.hipstr_output['viz.gz']} \\
+            --log {hipstr_job.hipstr_output['log.txt']} \\
+            --output-filters
+        """
+        )
+        # HipSTR output writing
+        hipstr_output_path_vcf = output_path(
+            f'{output_file_name}_shard{i+1}.vcf.gz', 'analysis'
+        )
+        b.write_output(hipstr_job.hipstr_output['vcf.gz'], hipstr_output_path_vcf)
+
+        hipstr_output_path_viz = output_path(
+            f'{output_file_name}_shard{i+1}.viz.gz', 'analysis'
+        )
+        b.write_output(hipstr_job.hipstr_output['viz.gz'], hipstr_output_path_viz)
+
+        hipstr_output_path_log = output_path(
+            f'{output_file_name}_shard{i+1}.log.txt', 'analysis'
+        )
+        b.write_output(hipstr_job.hipstr_output['log.txt'], hipstr_output_path_log)
```

Comment:
I think you can rewrite this whole section as
'''
b.write_output(hipstrjob.hipstroutput, f'...')
'''

This will copy the whole output group out to the same root, with all the correct extensions as defined in the resource group